File: src/main/kotlin/com/iodigital/weather/api/WeatherAPIResponse.java
package com.iodigital.weather.api;

import com.iodigital.weather.WeatherInfo;

import java.util.List;

public record WeatherAPIResponse(Location location, Forecast forecast) {
    public List<WeatherInfo> toWeatherInfoList() {
        final var region = location.region();
        final var country = location.country();
        final var city = location.city();
        return forecast.days().stream().map(f ->
                new WeatherInfo(
                        null,
                        region,
                        country,
                        "",
                        city,
                        f.date(),
                        "%.2f".formatted(f.temperature().avgF())
                )
        ).toList();
    }
}

File: src/main/kotlin/com/iodigital/weather/api/WeatherAPIResponse.kt
package com.iodigital.weather.api

import com.iodigital.weather.WeatherInfo

@JvmRecord
data class WeatherAPIResponse(val location: Location, val forecast: Forecast) {
    fun toWeatherInfoList(): List<WeatherInfo> =
        forecast.days.map {
            WeatherInfo(
                id = null,
                region = location.region,
                country = location.country,
                state = "",
                city = location.city,
                localDate = it.date,
                avgTemperature = "%.2f".format(it.temperature.avgF)
            )
        }
}
